### ISSUE TITLE
Cannot access http://fritz.box/webservices/homeautoswitch.lua

### ISSUE THREAD

**Issue by atomikx** [andig/pimatic-fritz#34]
Hello,

I want to controll a FrizDECT 200 switch with pimatic. After i install the fritz plugin in my pimatic v0.9.37 i get the folowing error messages in the pimatic-daemon.log:

- error [pimatic-fritz]: Cannot access http://fritz.box/webservices/homeautoswitch.lua?0=0&switchcmd=getswitchlist: undefined

- Unhandled rejection (<{"error":{"code":"ECONNRESET"},"option...>, no stack trace)

- [pimatic-fritz] warn: Re-establishing session at http://fritz.box

Despite the errror message i can switch the DECT 200, see the temperature an can switch WLAN on/off.

On my FritzBox 7390 i use FRITZ!OS:06.51. Login with an explicit FB-User ("_pimatic_") with all selectable rights. The Fritz-Box-Log says "_Anmeldung des Benutzers pimatic an der FRITZ!Box Benutzeroberfläche von IP-Adresse xxx.xxx.xxx.xxx_".

For pimatic i use a Raspberry Pi Model B+ with Raspbian GNU/Linux 8 (jessie)
Arduino Compatible Nano V3
Node JS v4.4.5
Pimatic v0.0.37
Fritz Plugin v0.5.3

Best regards 
Atomikx

**Comment by atomikx** [comment]
to complement my informations, here are my configuration settings (config.json):

```
  "plugins": [
	...
    {
      "url": "http://fritz.box",
      "user": "pimatic",
      "password": "xxxxxxxx",
      "plugin": "fritz",
      "active": true
    }
  ],
  "devices": [
    {
      "id": "home-switch",
      "name": "Fritz Steckdose",
      "class": "FritzOutlet",
      "ain": "087620365940"
    },
    {
      "id": "temp-1",
      "name": "Temperatur Sensor 1",
      "class": "FritzTemperatureSensor",
      "ain": "087620365940"
    },
    {
      "id": "guest-wlan",
      "name": "Gäste WLAN",
      "class": "FritzWlan"
    },
	...
```

**Comment by andig** [comment]
Does the message appear only once after startup? Is there any functionality problem?

**Comment by atomikx** [comment]
These error messages appear once after startup. 
But now, after playing arround with the GuestWLAN switch i see more connection Errors on the pimatic site ("_Oops... no connection_"; sometimes "_timeout_"; sometimes: "_socket error_"). 

In the pimatic-daemon.log i found following Messages:
`
22:31:50.245 [pimatic-fritz] error: Cannot access http://fritz.box/webservices/homeautoswitch.lua?0=0&sid=bce3679a5fdacd07&switchcmd=gettemperature&ain=087610345930: undefined
Unhandled rejection (<{"error":{"code":"ENOTFOUND","errno":"...>, no stack trace)
22:30:30.150 [pimatic-fritz] error: Cannot access http://fritz.box/webservices/homeautoswitch.lua?0=0&sid=bce3679a5fdacd07&switchcmd=getswitchstate&ain=087610345930: undefined
Unhandled rejection (<{"error":{"code":"ENOTFOUND","errno":"...>, no stack trace)
`

**Comment by andig** [comment]
Different error? If Es please open new issue wihh steps to reproduce. The first one is just the initial login and should not be harmful.

**Comment by atomikx** [comment]
Okay, i made some more test.
I think there are realy two problems.

1) Error messages at pimatic start:
now i have put the plugin configuration to first position in der plugin section (config.json) -> there are no more error messages in the log file now

2) Switching on, off, on, ... the Guest WLAN-Switch show "disconnect" messages on the pimatic web-interface and log file (see above):
This happens only in WLAN Mode (no, the raspi ist not connectet to the guest-wlan;-) - if the raspi conneted to LAN the switch works fine.
I can open a separate issue in few days

**Comment by atomikx** [comment]
concerning to 1 i forgot to say that i must remove the guest-wlan device ("FritzWlan")in the config file. Then all error messages   are away and the AINs can get from the Fritzbox.

**Comment by andig** [comment]
I can only imagine that switching guest wlan briefly interrupts normal wlan, too. I have really no idea though if that is the case.

**Comment by andig** [comment]
Closing due to inactivity
